code/datums/diseases/_MobProcs.dm

```diff
@@ -147,6 +147,11 @@
 
 
 /mob/living/carbon/human/CanContractDisease(datum/disease/D)
-	if(dna && (VIRUSIMMUNE in dna.species.species_traits))
+	if(dna && (VIRUSIMMUNE in dna.species.species_traits) && !D.bypasses_immunity)
 		return 0
+	if(length(D.required_organs))
+		for(var/thing in D.required_organs)
+			if(!locate(thing) in bodyparts)
+				if(!locate(thing) in internal_organs)
```

Comment:
Should be if(!((locate(thing) in bodyparts) || (locate(thing) in internal_organs))